File: persistence/persistence.js
const persistence = (number) => {
    if(number < 9) return 0;
  
    let count = 0;
    let multiply = number;
    do {
        count++;
        multiply = multiply.toString().split('');
        multiply = multiply.reduce((acc, curr) => {
            return acc * curr;
        });
    } while (multiply > 9);

    return count; 
};

module.exports = { persistence };

//https://www.codewars.com/kata/55bf01e5a717a0d57e0000ec/train/javascript

File: python-venv/persistence/persistence.py
from functools import reduce
def persistence(number):

    count = 0
    multiply = number

    while multiply > 9:
        count += 1
        multiply = list(str(multiply))
        multiply = reduce((lambda a, b: int(a) * int(b)), multiply)

    return count

#https://www.codewars.com/kata/55bf01e5a717a0d57e0000ec/train/python